ui/app/models/secret-engine.js

```diff
@@ -22,127 +22,194 @@ const validations = {
-  }),
-  casRequired: attr('boolean', {
+  })
+  maxVersions;
+  @attr('boolean', {
     defaultValue: false,
     label: 'Require Check and Set',
     subText:
       'If checked, all keys will require the cas parameter to be set on all write requests. A key’s metadata settings can overwrite this value.',
-  }),
-  deleteVersionAfter: attr({
+  })
+  casRequired;
+  @attr({
     defaultValue: 0,
     editType: 'ttl',
     label: 'Automate secret deletion',
     helperTextDisabled: 'A secret’s version must be manually deleted.',
     helperTextEnabled: 'Delete all new versions of this secret after',
-  }),
+  })
+  deleteVersionAfter;
+
+  /* GETTERS */
+  get modelTypeForKV() {
+    const engineType = this.engineType;
+    if ((engineType === 'kv' || engineType === 'generic') && this.version === 2) {
+      return 'secret-v2';
+    }
+    return 'secret';
+  }
+  get isV2KV() {
+    return this.modelTypeForKV === 'secret-v2';
+  }
+
+  get attrs() {
+    return this.formFields.map((fieldName) => {
+      return this.allByKey[fieldName];
+    });
+  }
+
+  get fieldGroups() {
+    return this.formFieldGroups.map((group) => {
+      // Each group object has 1 key (name of group) with array value
+      const groupName = Object.keys(group)[0];
+      const expanded = group[groupName]?.map((fieldName) => this.allByKey[fieldName]);
+      return { [groupName]: expanded };
+    });
+  }
 
-  modelTypeForKV: computed('engineType', 'version', function () {
-    const type = this.engineType;
-    let modelType = 'secret';
-    if ((type === 'kv' || type === 'generic') && this.version === 2) {
-      modelType = 'secret-v2';
+  get icon() {
+    if (!this.engineType || this.engineType === 'kmip') {
+      return 'secrets';
     }
-    return modelType;
-  }),
+    if (this.engineType === 'keymgmt') {
+      return 'key';
+    }
+    return this.engineType;
+  }
+
+  get engineType() {
+    return (this.type || '').replace(/^ns_/, '');
+  }
 
-  isV2KV: equal('modelTypeForKV', 'secret-v2'),
+  get shouldIncludeInList() {
+    return !LIST_EXCLUDED_BACKENDS.includes(this.engineType);
+  }
+
+  get localDisplay() {
+    return this.local ? 'local' : 'replicated';
+  }
 
-  formFields: computed('engineType', 'version', function () {
+  get formFields() {
     const type = this.engineType;
     const fields = ['type', 'path', 'description', 'accessor', 'local', 'sealWrap'];
     // no ttl options for keymgmt
-    const ttl = type !== 'keymgmt' ? 'defaultLeaseTtl,maxLeaseTtl,' : '';
+    if (type !== 'keymgmt') {
+      fields.push('config.defaultLeaseTtl', 'config.maxLeaseTtl');
+    }
     fields.push(
-      `config.{${ttl}auditNonHmacRequestKeys,auditNonHmacResponseKeys,passthroughRequestHeaders,allowedResponseHeaders}`
+      'config.auditNonHmacRequestKeys',
```

Comment:
Seeing this kind of stuff makes me so badly want to move each secret engine to it's own engine with their own models/adapters, etc.  They're all unique in their own ways.